### PR TITLE
TST: Add SparseArray _arith_method test to fail on incorrect comparison operator

### PR DESCRIPTION
**Pull request by jsngn**
- [x] Part of #58517 
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] ~Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.~ Not applicable
- [ ] ~Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.~ Not applicable

Before, if the line [core/arrays/sparse/array.py:1768](https://github.com/pandas-dev/pandas/blob/v2.2.1/pandas/core/arrays/sparse/array.py#L1768) is changed from  `if len(self) != len(other):` to `if len(self) > len(other):`, all tests still passed. Now they will fail if the comparison operator is changed to anything besides `!=`.

### CODE REVIEW

**Comment by jsngn** [comment]
@mroeschke Could you take a look at the change? Thanks!

File: pandas/tests/arrays/sparse/test_arithmetics.py [hunk 1]
```diff
@@ -468,6 +468,19 @@ def test_mismatched_length_cmp_op(cons):
         left & right
 
 
+@pytest.mark.parametrize(
+    "a, b",
+    [
+        (SparseArray([0, 1, 2], fill_value=0), np.array([0, 1, 2, 3])),
```
Comment (mroeschke) [inline pandas/tests/arrays/sparse/test_arithmetics.py:474]: Nit: Could you move the `SparseArray` and `np.array` call into the test body?

Comment (jsngn) [reply pandas/tests/arrays/sparse/test_arithmetics.py]: Done

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @jsngn

**Pull request merged**